lib/remoteResource.js

```diff
@@ -26,16 +26,71 @@ const requestsTemplate = `{
   }
 }`;
 
+const requestsTemplateGit = `{
+  "options": {
+    "git": {
+      "provider": "{{{provider}}}",
+      "repo": "{{{repo}}}",
+      "branch": "{{{branch}}}",
+      "filePath": "{{{filePath}}}"
+    },
+    "headers": {
+      "razee-org-key": {
+        "valueFrom": {
+          "secretKeyRef":{
+            "name": "{{{secretName}}}",
+            "namespace": "{{namespace}}",
+            "key": "razee-api-org-key"
+          }
+        }
+      },
+      "Authorization": {
+        "valueFrom": {
+          "secretKeyRef":{
+            "name": "{{{secretNameGit}}}",
+            "namespace": "{{namespace}}",
+            "key": "token"
+          }
+        }
+      }
+    }
+    
+  }
+}`; 
+
 const createRemoteResources = async (razeeApi, apiKey, subscriptions, clusterId) => {
   log.info('create remote resources subscription list', { subscriptions });
   try {
     return Promise.all(subscriptions.map(async sub => {
       const apiKeyBase64 = Buffer.from(apiKey).toString('base64');
       const url = `${razeeApi}/${sub.url}`;
       const secretName = `clustersubscription-${sub.subscriptionUuid}-secret`;
-      const rendered = Mustache.render(requestsTemplate, { url: url, secretName: secretName, namespace: NAMESPACE });
+      let rendered;
+      let backendService;
+      let secretNameGit = 'satellite-config-gitops';
+      if (sub.url) {
+        rendered = Mustache.render(requestsTemplate, { url: url, secretName: secretName, namespace: NAMESPACE });
+        backendService = 'generic';
+      } else if (sub.remote) {
+        const provider = sub.remote.remoteType;
+        let repo;
+        let branch;
+        let filePath;
+        sub.remote.parameters.map((param) => {
+          if (param.key == 'repo') {
+            repo = param.value;
+          } else if (param.key == 'branch') {
+            branch = param.value;
+          } else if (param.key == 'filePath') {
+            filePath = param.value;
+          } else if (param.key == 'secretName') {
+            secretNameGit = param.value;
+          }
```

Comment:
```suggestion
        for( const param of sub.remote.parameters ) {
          if (param.key == 'repo') {
            repo = param.value;
          } else if (param.key == 'branch') {
            branch = param.value;
          } else if (param.key == 'filePath') {
            filePath = param.value;
          } else if (param.key == 'secretName') {
            secretNameGit = param.value;
          }
```
`map()` isn't exactly the right approach here ("The map() method creates a new array populated with the results of calling a provided function on every element in the calling array").  Instead, try `sub.remote.parameters.forEach(...` or (as in suggestion above) `for( const param of sub.remote.parameters ) {...}`.

Reply:
.foreach(... is a good alternative